lazyflow/utility/reorderAxesDecorator.py

```diff
@@ -0,0 +1,263 @@
+import functools
+from lazyflow.operators.opReorderAxes import OpReorderAxes
+
+
+def reorder_options(internal_axes_order, exceptions=[],
+                    output_axes_order='tczyx'):
+    """
+        Adds reorder options to the specific operator class that the decorator
+        reorder accesses. This is not done by handing arguments to the reorder
+        decorator directly, because then the reorder decorator would need to be
+        wrapped, which would lead return a function (wrapper) as the operator
+        class, which conflicts with inheritance of the specific operator (as
+        the inheriting operator class appears to be a function). Therefore the
+        arguments are outsourced in this decorator.
+
+        note: Use this function decorator only in combination with the reorder
+              decorator and as second (inner) decorator.
+              (see example usage below)
+
+        @param internal_axes_order: str Axes order the decorated operator
+                                        assumes internally.ResizedShape
+        @param exceptions: list(str) List of slot names that do not need to be
+                                     reordered.
+        @param output_axes_order: str Output Axes order for all reordered
+                                      output slots.
+
+        Example usage:
+
+            @reorder
+            @reorder_options('tzyxc', ['ResizedShape'])
+            class OpResize5D(Operator):
+                ...
+
+    """
+    assert isinstance(internal_axes_order, str)
+    assert isinstance(exceptions, list)
+    assert all([isinstance(ex, str) for ex in exceptions])
+    assert isinstance(output_axes_order, str)
+
+    def add_exceptions(cls):
+        assert all([ex in [s.name for s in cls.inputSlots + cls.outputSlots]
+                    for ex in exceptions])
+        cls._internal_axes_order = internal_axes_order
+        cls._reorder_exceptions = exceptions
+        cls._output_axes_order = output_axes_order
+        return cls
+
+    return add_exceptions
+
+
+def guard_methods(cls):
+    """
+        helper function for reorder
```

Comment:
I think you went pretty deep into the internals with the code in this file. It would be nice if you could share the gained knowledge with some more comments. For me it's not easy to follow the code and understand the motivations. Maybe it would be good to get some explanations in person